WordPress/Classes/ViewRelated/Views/AlertView/AlertView.swift

```diff
@@ -0,0 +1,122 @@
+    public typealias Completion = (() -> ())
+
+    
+    /**
+    *  @details     Designated Initializer
+    *  @param       title       The title of the AlertView.
+    *  @param       message     Message string to be displayed. Note: Bold is supported, **markdown flavor**.
+    *  @param       completion  A closure to be executed right after the button is pressed.
+    */
+    public init(title: String, message: String, button: String, completion: Completion?) {
+        super.init()
+        
+        // Load the nib
+        NSBundle.mainBundle().loadNibNamed(AlertView.classNameWithoutNamespaces(), owner: self, options: nil)
+        
+        // Check the Outlets
+        assert(internalView != nil)
+
+        // Done!
+        internalView.titleLabel.text = title
+        internalView.descriptionLabel.attributedText = applyMessageStyles(message)
+    }
+    
+    
+    
+    /**
+    *  @details     Displays the AlertView in the window.
+    */
+    public func show() {
+        let targetView = keyView()
+
+        // Attach the BackgroundView
+        targetView.endEditing(true)
+        targetView.addSubview(internalView)
+        
+        // We should cover everything
+        internalView.setTranslatesAutoresizingMaskIntoConstraints(false)
+        targetView.pinSubviewToAllEdges(internalView)
+        
+        // Animate!
+        internalView.fadeInWithAnimation()
+        
+        // Note:
+        // The internalView will retain the AlertView itself. After it's dismissed, everything will
+        // be automatically cleaned up!
+        internalView?.onClick = {
+            self.dismiss()
+        }
+    }
+    
+    
+    
+    /**
+    *  @details     Finds the root view attached to the window
+    *  @returns     The Key View.
+    */
+    private func keyView() -> UIView {
+        return UIApplication.sharedApplication().keyWindow?.subviews.first as! UIView
+    }
+    
+    
+    
+    /**
+    *  @details     Fades out the AlertView, and, on completion, will cleanup and hit the completion closure.
+    */
+    private func dismiss() {        
+        UIView.animateWithDuration(WPAnimationDurationFast, animations: { () -> Void in
+                self.internalView.alpha = WPAlphaZero
+            },
+            completion: { (success: Bool) -> Void in
+                self.internalView.removeFromSuperview()
+                self.onCompletion?()
+            })
+    }
+    
+    
+    
+    /**
+    *  @details     Apples Bold Style over all of the text surrounded by **double stars** (and removes the markers!).
+    *  @param       message     The Message that should be stylized.
+    *  @returns                 The Message with Bold Substrings styled.
+    */
+    private func applyMessageStyles(message: String) -> NSAttributedString {
+        let attributedMessage = NSMutableAttributedString(string: message, attributes: Style.detailsRegularAttributes)
+        
+        // Apply Bold Styles
```

Comment:
Following up a bit on comments... in some cases like here, comments can be replaced more gracefully with methods.

For example, you could replace all of the code + comments in this method with:

```swift
self.applyBoldStyle()
self.replaceBoldMarkers()
```

No biggies though, just my 2 cents.